### PR TITLE
Msue 41

### PR DESCRIPTION
**Pull request by sabitha-varghese**
- Verfication endpoints added

### CODE REVIEW

File: lib/SiftClient.php [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 class SiftClient {
-    const API_ENDPOINT = 'https://api.sift.com';
 
+    const API_ENDPOINT = 'https://api.sift.com';
     // Must be kept in sync with composer.json
     const API_VERSION = '205';
-
     const API3_VERSION = '3';
-
+    const API4_VERSION = '1.1';
```
Comment (mlunney) [inline lib/SiftClient.php:9]: It's not API4. Verification API has its own version numbers.

File: lib/SiftClient.php [hunk 2]
```diff
@@ -130,19 +129,23 @@ public function track($event, $properties, $opts = array()) {
...
             $request = new SiftRequest(
-                $path, SiftRequest::POST, $opts['timeout'], $opts['version'], array(
-                    'body' => $properties,
-                    'params' => $params
-                ),
-                $this->curl_opts
+                    $path, SiftRequest::POST, $opts['timeout'], $opts['version'], array(
+                'body' => $properties,
+                'params' => $params
+                    ), $this->curl_opts
```
Comment (mlunney) [inline lib/SiftClient.php:148]: I think the old indenting was better here.

File: lib/SiftClient.php [hunk 3]
```diff
@@ -737,20 +729,19 @@ private function applyDecision($url, $opts = array()) {
...
-                $opts['timeout'],
-                self::API3_VERSION,
-                array(
-                    'auth' => $this->api_key . ':',
-                    'body' => $body
-                )
+                    $url, SiftRequest::POST, $opts['timeout'], self::API3_VERSION, array(
+                'auth' => $this->api_key . ':',
+                'body' => $body
+                    )
```
Comment (mlunney) [inline lib/SiftClient.php:744]: old indentation was easier to read.

File: lib/SiftClient.php [hunk 4]
```diff
@@ -760,6 +751,105 @@ private function applyDecision($url, $opts = array()) {
...
+            'version' => self::API4_VERSION,
+        ));
+
+        $this->validateArgument($parameters['$user_id'], 'user id', 'string');
+        $curl_opts[CURLOPT_HTTPHEADER] = array('Authorization: Basic ' . base64_encode($this->api_key . ':'));
+        try {
+            $request = new SiftRequest(
+                    self::userSendApiUrl($opts['version']), SiftRequest::POST, $opts['timeout'], $opts['version'], array(
+                'body' => $parameters,
+                    ), $curl_opts
```
Comment (mlunney) [inline lib/SiftClient.php:810]: indentation looks strange here.

File: lib/SiftClient.php [hunk 5]
```diff
@@ -760,6 +751,105 @@ private function applyDecision($url, $opts = array()) {
...
+     *
+     * @param array $opts  Array of optional parameters for this request:
+     *     - int 'timeout': By default, this client's timeout is used.
+     *     - string 'version': By default, this client's version is used.
+     *
+     * @return null|SiftResponse
+     */
+    public function send($parameters, $opts = array()) {
+        $this->mergeArguments($opts, array(
+            'abuse_types' => array(),
```
Comment (mlunney) [inline lib/SiftClient.php:799]: What's `abuse_types` for in send and resend?

File: test/SiftClientTest.php [hunk 6]
```diff
@@ -658,10 +658,41 @@ public function testApplyDecisionToContentWithSpecialCharacters() {
...
-        SiftRequest::setMockResponse($mockUrl, SiftRequest::POST ,$mockResponse);
+        SiftRequest::setMockResponse($mockUrl, SiftRequest::POST, $mockResponse);
 
         $response = $this->client->track('$create_content', $this->profile_properties);
         $this->assertTrue($response->isOk());
         $this->assertEquals($response->apiErrorMessage, 'OK');
     }
+
+    public function testSend() {
+        $mockUrl = 'https://api.sift.com/v1.1/verification/send';
```
Comment (mlunney) [inline test/SiftClientTest.php:669]: Did these tests pass with the version being wrong in the actual send function? I think it should have caught the bug in the test cases when you ran them.

Comment (sabitha-varghese) [reply test/SiftClientTest.php:669]: Yes..It was my mistake. sorry!

**Review by mlunney — APPROVED** [review]
LGTM

**Pull request merged**